account_bank_statement_import_coda/wizard/account_bank_statement_import_coda.py

```diff
@@ -0,0 +1,218 @@
+        if not self._check_coda(data_file):
+            return super(account_bank_statement_import, self)._parse_file(
+                data_file)
+        vals_bank_statements = []
+        try:
+            statements = Parser().parse(data_file)
+            for statement in statements:
+                vals_bank_statements.append(
+                    self.get_st_vals(statement))
+        except Exception, e:
+            _logger.exception('Error when parsing coda file')
+            raise Warning(
+                _("The following problem occurred during import. "
+                  "The file might not be valid.\n\n %s" % e.message))
+
+        acc_number = None
+        currency = None
+        if statements:
+            acc_number = statements[0].acc_number
+            currency = statement.currency
+        return currency, acc_number, vals_bank_statements
+
+    def get_st_vals(self, statement):
+        """
+        This method return a dict of vals that can be passed to
+        create method of statement.
+        :return: dict of vals that represent additional infos for the statement
+            found in the file.
+            {
+             'name': paper_seq_number
+             'balance_start': balance_start,
+             'balance_end_real': balance_end_real,
+             'transactions': transactions
+            }
+        """
+        balance_start = statement.old_balance
+        if statement.old_balance_amount_sign == AmountSign.DEBIT:
+            balance_start = - balance_start
+        balance_end_real = statement.new_balance
+        if statement.new_balance_amount_sign == AmountSign.DEBIT:
+            balance_end_real = - balance_end_real
+        transactions = []
+        vals = {'balance_start': balance_start,
+                'balance_end_real': balance_end_real,
+                'date': statement.creation_date,
+                'transactions': transactions
+                }
+        name = statement.paper_seq_number
+        if name:
+            year = ""
+            if statement.creation_date:
+                parsed_date = date_parser.parse(statement.creation_date)
+                year = "%s/" % parsed_date.year
+            vals.update({
+                'name': "%s%s" % (year, statement.paper_seq_number),
+            })
+
+        globalisation_dict = dict([
+            (st.ref_move, st) for st in statement.movements
+            if st.type == MovementRecordType.GLOBALISATION])
+        for sequence, line in enumerate(
+                filter(lambda l: l.type != MovementRecordType.GLOBALISATION,
+                       statement.movements)
+                ):
+            if line.type != MovementRecordType.GLOBALISATION:
+                info = self.get_st_line_vals(line,
+                                             globalisation_dict)
+                info['sequence'] = sequence
+                transactions.append(info)
+        coda_notes = []
+        for line in statement.informations:
```

Comment:
In this module, 'line' has many different meanings at different places. It makes the code a bit difficult to understand.